File: RepoTransBench/source_projects/Java/fafaldo-zz_FABToolbar/FABToolbar/library/src/test/java/com/github/fafaldo/fabtoolbar/widget/FABToolbarLayoutPublicTest.java
package com.github.fafaldo.fabtoolbar.widget;

import android.content.Context;
import android.content.res.TypedArray;
import android.util.AttributeSet;

import org.junit.Before;
import org.junit.Test;

import static org.junit.Assert.*;
import static org.mockito.Mockito.*;

public class FABToolbarLayoutPublicTest {

    private Context mockContext;
    private AttributeSet mockAttrs;
    private TypedArray mockTypedArray;

    @Before
    public void setup() {
        mockContext = mock(Context.class);
        mockAttrs = mock(AttributeSet.class);
        mockTypedArray = mock(TypedArray.class);

        // Use different mock return values than in the original test
        when(mockContext.obtainStyledAttributes(any(AttributeSet.class), any(int[].class)))
                .thenReturn(mockTypedArray);

        when(mockTypedArray.getInt(anyInt(), anyInt())).thenReturn(888);
        when(mockTypedArray.getDimensionPixelSize(anyInt(), anyInt())).thenReturn(50);
        when(mockTypedArray.getFloat(anyInt(), anyFloat())).thenReturn(0.88f);
        when(mockTypedArray.getResourceId(anyInt(), anyInt())).thenReturn(42);
        when(mockTypedArray.getBoolean(anyInt(), anyBoolean())).thenReturn(false);
    }

    @Test
    public void testConstructorsWithOtherValues() {
        // Context constructor
        FABToolbarLayout layout1 = new FABToolbarLayout(mockContext);
        assertNotNull(layout1);

        // Context + AttributeSet constructor
        FABToolbarLayout layout2 = new FABToolbarLayout(mockContext, mockAttrs);
        assertNotNull(layout2);

        // Context + AttributeSet + defStyleAttr constructor
        FABToolbarLayout layout3 = new FABToolbarLayout(mockContext, mockAttrs, 1);
        assertNotNull(layout3);
    }

    @Test
    public void testParseAttrsPublic() {
        new FABToolbarLayout(mockContext, mockAttrs);
        verify(mockContext, atLeastOnce()).obtainStyledAttributes(any(AttributeSet.class), any(int[].class));
        verify(mockTypedArray, atLeastOnce()).recycle();
    }
}

File: RepoTransBench/target_projects/Java/C#/fafaldo-zz_FABToolbar/public_tests/widget/FABToolbarLayoutPublicTest.cs
using System;
using Moq;
using Xunit;
using FAFToolbar.Widget;

namespace PublicTests.Widget
{
    public class FABToolbarLayoutPublicTest
    {
        private Mock<IContext> mockContext;
        private Mock<IAttributeSet> mockAttrs;
        private Mock<ITypedArray> mockTypedArray;

        public FABToolbarLayoutPublicTest()
        {
            SetUp();
        }

        public void SetUp()
        {
            mockContext = new Mock<IContext>();
            mockAttrs = new Mock<IAttributeSet>();
            mockTypedArray = new Mock<ITypedArray>();

            mockContext.Setup(c => c.ObtainStyledAttributes(It.IsAny<IAttributeSet>(), It.IsAny<int[]>()))
                .Returns(mockTypedArray.Object);

            mockTypedArray.Setup(t => t.GetInt(It.IsAny<int>(), It.IsAny<int>())).Returns(888);
            mockTypedArray.Setup(t => t.GetDimensionPixelSize(It.IsAny<int>(), It.IsAny<int>())).Returns(50);
            mockTypedArray.Setup(t => t.GetFloat(It.IsAny<int>(), It.IsAny<float>())).Returns(0.88f);
            mockTypedArray.Setup(t => t.GetResourceId(It.IsAny<int>(), It.IsAny<int>())).Returns(42);
            mockTypedArray.Setup(t => t.GetBoolean(It.IsAny<int>(), It.IsAny<bool>())).Returns(false);
        }

        [Fact]
        public void TestConstructorsWithOtherValues()
        {
            // Context constructor
            var layout1 = new FABToolbarLayout(mockContext.Object);
            Assert.NotNull(layout1);

            // Context + AttributeSet constructor
            var layout2 = new FABToolbarLayout(mockContext.Object, mockAttrs.Object);
            Assert.NotNull(layout2);

            // Context + AttributeSet + defStyleAttr constructor
            var layout3 = new FABToolbarLayout(mockContext.Object, mockAttrs.Object, 1);
            Assert.NotNull(layout3);
        }

        [Fact]
        public void TestParseAttrsPublic()
        {
            var layout = new FABToolbarLayout(mockContext.Object, mockAttrs.Object);
            mockContext.Verify(c => c.ObtainStyledAttributes(It.IsAny<IAttributeSet>(), It.IsAny<int[]>()), Times.AtLeastOnce());
            mockTypedArray.Verify(t => t.Recycle(), Times.AtLeastOnce());
        }
    }

    // Mocks/stubs for public test context
    public interface IContext
    {
        ITypedArray ObtainStyledAttributes(IAttributeSet attrs, int[] attrs2);
    }

    public interface IAttributeSet { }

    public interface ITypedArray
    {
        int GetInt(int index, int defValue);
        int GetDimensionPixelSize(int index, int defValue);
        float GetFloat(int index, float defValue);
        int GetResourceId(int index, int defValue);
        bool GetBoolean(int index, bool defValue);
        void Recycle();
    }
}